config/goglog/goglog.go

```diff
@@ -38,165 +55,206 @@ func newLogger() *LoggerType {
 			Hooks:     make(logrus.LevelHooks),
 			Level:     logrus.InfoLevel,
 		},
+		WarnSentryHub:  warnHub,
+		ErrorSentryHub: errorHub,
+		FatalSentryHub: fatalHub,
 	}
 }
 
-// LoggerType wrap logrus.Logger type
-type LoggerType struct {
-	stdout *logrus.Logger
-	stderr *logrus.Logger
+// configureHub clones the current Sentry hub and sets the specified level
+func configureHub(level sentry.Level) *sentry.Hub {
+	hub := sentry.CurrentHub().Clone()
+	hub.ConfigureScope(func(scope *sentry.Scope) {
+		scope.SetLevel(level)
+	})
+	return hub
+}
+
+// LogWarning logs a warning message to Sentry
+func (l *LoggerType) LogWarning(msg string) {
+	l.WarnSentryHub.WithScope(func(scope *sentry.Scope) {
+		sentry.CaptureMessage(msg)
+	})
+}
+
+// LogError logs an error message to Sentry
+func (l *LoggerType) LogError(msg string) {
+	l.ErrorSentryHub.WithScope(func(scope *sentry.Scope) {
+		sentry.CaptureMessage(msg)
+	})
+}
+
+// LogFatal logs a fatal error message to Sentry and exits the program
+func (l *LoggerType) LogFatal(msg string) {
+	l.FatalSentryHub.WithScope(func(scope *sentry.Scope) {
+		sentry.CaptureMessage(msg)
+	})
+	logrus.Fatal(msg)
 }
 
 // WithField wrap logrus function
-func (t LoggerType) WithField(key string, value any) *logrus.Entry {
+func (t *LoggerType) WithField(key string, value any) *logrus.Entry {
 	return t.stdout.WithField(key, value)
 }
 
 // WithFields wrap logrus function
-func (t LoggerType) WithFields(fields logrus.Fields) *logrus.Entry {
+func (t *LoggerType) WithFields(fields logrus.Fields) *logrus.Entry {
 	return t.stdout.WithFields(fields)
 }
 
 // WithError wrap logrus function
-func (t LoggerType) WithError(err error) *logrus.Entry {
+func (t *LoggerType) WithError(err error) *logrus.Entry {
 	return t.stdout.WithError(err)
 }
 
 // Debugf wrap logrus function
-func (t LoggerType) Debugf(format string, args ...any) {
+func (t *LoggerType) Debugf(format string, args ...any) {
 	t.stdout.Debugf(format, args...)
 }
 
 // Infof wrap logrus function
-func (t LoggerType) Infof(format string, args ...any) {
+func (t *LoggerType) Infof(format string, args ...any) {
 	t.stdout.Infof(format, args...)
 }
 
 // Printf wrap logrus function
-func (t LoggerType) Printf(format string, args ...any) {
+func (t *LoggerType) Printf(format string, args ...any) {
 	t.stdout.Printf(format, args...)
 }
 
 // Warnf wrap logrus function
-func (t LoggerType) Warnf(format string, args ...any) {
+func (t *LoggerType) Warnf(format string, args ...any) {
```

Comment:
Why pointer?

Reply:
LoggerType is a large struct, copying it on each method call can be expensive. Using a pointer receiver avoids this overhead.